bdk_keychain/src/keychain_txout_index.rs

```diff
@@ -165,7 +166,8 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     pub fn store_all_up_to(&mut self, keychains: &BTreeMap<K, u32>) -> bool {
         keychains
             .into_iter()
-            .any(|(keychain, index)| self.store_up_to(keychain, *index))
+            .map(|(keychain, index)| self.store_up_to(keychain, *index))
+            .fold(false, |acc, v| acc || v)
```

Comment:
Thanks for finding this. I think it would be better style to do this is with a for loop and a bool variable.

Reply:
I like this style tbh.. 😄 Quite cool and compact..

Reply:
It was even more cool and compact before and it was a bug.